std/shared/src/main/scala/collection/Set.scala

```diff
@@ -0,0 +1,19 @@
+package scalaz
+package std
+
+import typeclass.{ EqClass, MonadClass }
+
+trait SetInstances {
+  implicit val setMonad: Monad[Set] = instanceOf(new MonadClass[Set] {
```

Comment:
eww. how principled are we going to be about the stdlib instances?

Reply:
So from what I read, a set has a monad if its `A` has an `Ord`... I don't know that I can represent that. So may be better to just remove Set?

Reply:
Yeah. The generalized form is

```scala 
trait RFunctor[F[_]] {
  type Constr[T]
  def rfmap[A, B: Constr](fa: F[A])(f: A => B): F[B]
}
```

and so on. (You get `trait Functor[F[_]] extends RFunctor[F] { type Constr[T] = DummyImplicit }`)

So either we include the generalizations like this (functors into subcategories IIRC) or forego this instance.

Reply:
I removed it :D